config/initializers/0_ldap_authenticatable.rb

```diff
@@ -0,0 +1,109 @@
+require 'net/ldap'
+require 'devise/strategies/authenticatable'
+
+module Devise
+  module Strategies
+    class LdapAuthenticatable < Authenticatable
+      def valid?
+        Rails.configuration.x.use_ldap && params[:user].present?
+      end
+
+      def authenticate!
+        config = Rails.configuration
```

Comment:
maybe a better way to do this would be something like `delegate :ldap_host, :ldap_port, :ldap_manager_dn, :ldap_manager_password, :ldap_manager_bind, :ldap_search_filter, :ldap_search_base, :ldap_mail_attribute, :ldap_username_attribute, to: Rails.configuration.x`, or something?

Honestly, it doesn't feel like Rails.configuration.x is the right place to put these variables...